File: src/baseDatos/Deserializador.java
package baseDatos;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.util.ArrayList;

import gestorAplicacion.gestorMusica.Cancion;
import gestorAplicacion.gestorMusica.Coleccion;
import gestorAplicacion.gestorMusica.Lista;
import gestorAplicacion.gestorMusica.MeGusta;
import gestorAplicacion.gestorPersonas.Artista;
import gestorAplicacion.gestorPersonas.Usuario;

public class Deserializador {
	
	public static <E> void deserializar(ArrayList<E> lista, String nombre) {
		
		FileInputStream fileIn;
		
		try {
			String path = System.getProperty("user.dir") + "/practica-g1-equipo-3/src/baseDatos/temp/" + nombre + ".txt";
			// System.out.println("Ruta " + path);
			fileIn = new FileInputStream(path);
			ObjectInputStream in = new ObjectInputStream(fileIn);
			ArrayList<E> listaNueva = (ArrayList<E>) in.readObject();
			// System.out.println(listaNueva);
			
			for (E elemento : listaNueva) {
				lista.add(elemento);
			}
			in.close();
			fileIn.close();

		} catch (FileNotFoundException e) {
			e.printStackTrace();
		} catch (IOException e) {
			System.out.println("El archivo está vacío");
		} catch (ClassNotFoundException e) {
			e.printStackTrace();
		}
	}
	
	public static void deserializarDatos() {
		Deserializador.deserializar(Usuario.getUsuariosExistentes(), "Usuarios");
		Deserializador.deserializar(Artista.getArtistasDisponibles(), "Artistas");
		Deserializador.deserializar(Cancion.getCancionesDisponibles(), "Canciones");
		Deserializador.deserializar(Coleccion.getColeccionesExistentes(), "Colecciones");
		Deserializador.deserializar(Lista.getListasExistentes(), "Listas");
		Deserializador.deserializar(MeGusta.getFavoritosExistentes(), "MeGusta");
	}
}

File: Python/src/baseDatos/deserializador.py
''' Módulo deserializador.py
    Autores: Carolina Álvarez Murillo, Miller Johan Chica Acero,
             Tomás Rodríguez Taborda, Jerónimo Ledesma Patiño,
             Catalina Restrepo Salgado
    Este módulo contiene la clase Deserializador, que nos va a permitir acceder al flujo de bytes que se
    haya guardado previamente en cada archivo .txt
'''

import pickle
import pathlib
import os

from gestorAplicacion.gestorMusica.cancion import Cancion
from gestorAplicacion.gestorMusica.coleccion import Coleccion
from gestorAplicacion.gestorMusica.lista import Lista
from gestorAplicacion.gestorMusica.meGusta import meGusta
from gestorAplicacion.gestorPersonas.artista import Artista
from gestorAplicacion.gestorPersonas.usuario import Usuario

class Deserializador:

    def deserializar(lista, nombre):
        ruta = os.path.join(pathlib.Path(__file__).parent.absolute(), "temp/" + nombre + ".txt")

        try:
            picklefile = open(ruta, 'rb')

        except:
            picklefile = open(ruta, 'x')
            picklefile = open(ruta, 'rb')

        if os.path.getsize(ruta) > 0:
            lista = pickle.load(picklefile)
        picklefile.close()

        return lista

    def deserializarDatos():
        Usuario.setUsuariosExistentes(Deserializador.deserializar(Usuario.getUsuariosExistentes(), "Usuarios"))
        Artista.setArtistasDisponibles(Deserializador.deserializar(Artista.getArtistasDisponibles(), "Artistas"))
        Cancion.setCancionesDisponibles(Deserializador.deserializar(Cancion.getCancionesDisponibles(), "Canciones"))
        Coleccion.setColeccionesExistentes(Deserializador.deserializar(Coleccion.getColeccionesExistentes(), "Colecciones"))
        Lista.setListasExistentes(Deserializador.deserializar(Lista.getListasExistentes(), "Listas"))
        meGusta.setFavoritosExistentes(Deserializador.deserializar(meGusta.getFavoritosExistentes(), "MeGusta")) 
        